### ISSUE TITLE
Update shown even though up to date

### ISSUE THREAD

**Issue by arogl** [chrisant996/clink#299]
I just installed 1.3.20 and when starting get the below

![image](https://user-images.githubusercontent.com/1115472/172987270-f0a183c3-dee3-4331-83f3-8abfb84b0237.png)

**Comment by chrisant996** [comment]
Thanks for reporting this. I'll take a look.

Can you make it happen again, and then upload the clink.log file here?

**Comment by arogl** [comment]
``` log
5aa0 start_logger              154 ---- 2022/06/10 15:31:09.097 -------------------------------------------------
5aa0 start_logger              160 Host process is 'cmd.exe' (pid 23200)
5aa0 start_logger              164 DLL path is 'C:\Program Files (x86)\clink'
5aa0 start_logger              178 Windows version 10.0.19044 (x64)
5aa0 start_logger              181 Clink version 1.3.20.59deed (x64)
5aa0 hook_setter::hook_setter  156 >>> Started hook transaction.
5aa0 hook_setter::attach_iat   313 Attempting to hook SetEnvironmentVariableW in IAT for module 00007FF71ECC0000.
5aa0 pe_info::iterate_imports  159 Found import in 'api-ms-win-core-processenvironment-l1-1-0.dll'
5aa0 find_iat                  121 Found import at 00007FF71ECF3860 (value is 00007FF942F2C540).
5aa0 hook_setter::attach_iat   313 Attempting to hook WriteConsoleW in IAT for module 00007FF71ECC0000.
5aa0 pe_info::iterate_imports  159 Found import in 'api-ms-win-core-console-l1-1-0.dll'
5aa0 find_iat                  121 Found import at 00007FF71ECF34F8 (value is 00007FF942ED1630).
5aa0 hook_setter::attach_iat   313 Attempting to hook GetEnvironmentVariableW in IAT for module 00007FF71ECC0000.
5aa0 pe_info::iterate_imports  159 Found import in 'api-ms-win-core-processenvironment-l1-1-0.dll'
5aa0 find_iat                  121 Found import at 00007FF71ECF3880 (value is 00007FF942EEABD0).
5aa0 hook_setter::commit       204 <<< Hook transaction committed.
5aa0 hook_setter::hook_setter  156 >>> Started hook transaction.
5aa0 hook_setter::detach_iat   356 Attempting to unhook 00007FF8E8B496A0 from GetEnvironmentVariableW in IAT for module 00007FF71ECC0000.
5aa0 pe_info::iterate_imports  159 Found import in 'api-ms-win-core-processenvironment-l1-1-0.dll'
5aa0 find_iat                  121 Found import at 00007FF71ECF3880 (value is 00007FF8E8B496A0).
5aa0 hook_setter::commit       204 <<< Hook transaction committed.
5aa0 hook_setter::hook_setter  156 >>> Started hook transaction.
5aa0 hook_setter::attach_iat   313 Attempting to hook ReadConsoleW in IAT for module 00007FF71ECC0000.
5aa0 pe_info::iterate_imports  159 Found import in 'api-ms-win-core-console-l1-1-0.dll'
5aa0 find_iat                  121 Found import at 00007FF71ECF34E0 (value is 00007FF942FC2BB0).
5aa0 hook_setter::commit       204 <<< Hook transaction committed.
5aa0 hook_setter::hook_setter  156 >>> Started hook transaction.
5aa0 hook_setter::attach_iat   313 Attempting to hook SetConsoleTitleW in IAT for module 00007FF71ECC0000.
5aa0 pe_info::iterate_imports  159 Found import in 'api-ms-win-core-console-l2-2-0.dll'
5aa0 find_iat                  121 Found import at 00007FF71ECF3558 (value is 00007FF942F3F490).
5aa0 hook_setter::commit       204 <<< Hook transaction committed.
5aa0 win_screen_buffer::begin  228 Using native terminal support; found 'Windows build >= 15063, console V2'.
5aa0 host_lua::load_scripts    138 Loaded 0 Lua scripts in 0 ms
5aa0 =?                          0 Clink updater: too soon to check for updates (1654833227 vs 1654839070).
5aa0 reset_handle              144 resetting mismatched stdout handle
5aa0 history_db::initialise   1249 master bank ctag: |CTAG_1616399996_110085875_163308_0
5aa0 history_db::compact      1494 History:  2696 active, 1363 deleted
5aa0 history_db::compact      1494 History:  2695 active, 1364 deleted
5aa0 reset_handle              144 resetting mismatched stderr handle
5aa0 history_db::compact      1494 History:  2694 active, 1365 deleted
```

![image](https://user-images.githubusercontent.com/1115472/172997314-1c71bfb7-deb9-4f46-9e98-d90eff010ab9.png)

**Comment by arogl** [comment]
What is the recommended way to update clink?

I normally download the install in case of re-install later

**Comment by chrisant996** [comment]
Well, starting in v1.3.18 Clink has a built in updater.

So that would be the recommended way of updating, now.

There's nothing wrong with also updating other ways, but there's currently a bug where if you haven't disabled the built in updater but you manually update instead of using the built in updater, then the built in updater gets confused into thinking there is still an update pending. It's just a bug, which will be fixed this weekend.

**Comment by chrisant996** [comment]
Fixed in v1.3.21.

**Comment by arogl** [comment]
@chrisant996 Thank you for your efforts in bringing clink up to date and the on-going fixes and improvements
